src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -81,9 +81,9 @@ public void Reset()
         // BEGIN: Structures that are safe to pool.
         private readonly HashSet<long> _dirtyPages;
         private readonly FastDictionary<long, long, NumericEqualityComparer> _dirtyOverflowPages;
-        private readonly Stack<long> _pagesToFreeOnCommit;
+        private readonly Stack<long> _pagesToFreeOnCommit; // TODO: This can be a regular list. There is no reason why we need to use an stack here. 
         private readonly FastDictionary<long, PageFromScratchBuffer, NumericEqualityComparer> _scratchPagesTable;
-        private readonly HashSet<PagerState> _pagerStates;
+        private readonly HashSet<PagerState> _pagerStates; // TODO: Understand if it makes sense for this to be a list instead. 
```

Comment:
Yes, but still if the number is small (say less than 10) it is way faster to just do that using a FastList behaving kind of as an LRU.